### PR TITLE
Minor improvements of AppState management in editoast

### PR DESCRIPTION
**Pull request by leovalais**
- **editoast: deprecate DbPool V1 and remove it from the AppState**
- **editoast: destructure AppState early in handlers for consistency**
- **editoast: fix minor destructuration inconsistencies**

> TIP
> Rapidly reviewable commit by commit

### CODE REVIEW

File: editoast/src/views/mod.rs [hunk 1]
```diff
@@ -334,8 +334,11 @@ async fn version() -> Json<Version> {
         (status = 200, description = "Return the core service version", body = Version),
     ),
 )]
-async fn core_version(app_state: State<AppState>) -> Json<Version> {
-    let core = app_state.core_client.clone();
+async fn core_version(
+    State(AppState {
+        core_client: core, ..
+    }): State<AppState>,
```
Comment (woshilapin) [inline editoast/src/views/mod.rs]: Is it worth it to `impl FromRef<AppState> for CoreClient`?

File: editoast/src/views/timetable.rs [hunk 2]
```diff
@@ -185,7 +185,7 @@ async fn post(
 async fn delete(
     State(db_pool): State<DbConnectionPoolV2>,
     Extension(auth): AuthenticationExt,
-    timetable_id: Path<TimetableIdParam>,
+    Path(TimetableIdParam { id }): Path<TimetableIdParam>,
```
Comment (woshilapin) [inline editoast/src/views/timetable.rs]: What do you think about keeping `timetable_id` for the rest of the code? (Note: I made only this comment, but this applies to a lot of occurences in the current commit).

```suggestion
    Path(TimetableIdParam { id: timetable_id }): Path<TimetableIdParam>,
```

**Review by woshilapin — APPROVED** [review]

File: editoast/src/views/timetable.rs (hunk 2 above)
Comment (leovalais) [reply editoast/src/views/timetable.rs]: Done in [`6968585` (#9935)](https://github.com/OpenRailAssociation/osrd/pull/9935/commits/6968585670931e0cb03bb27ed147424df3fde1df)

File: editoast/src/views/mod.rs (hunk 1 above)
Comment (leovalais) [reply editoast/src/views/mod.rs]: Done in [`fd2a3d9` (#9935)](https://github.com/OpenRailAssociation/osrd/pull/9935/commits/fd2a3d951594ed73d69fe6b39c411790d5638640)

**Comment by leovalais** [comment]
Thanks @woshilapin for the review! I'll let you resolve the conversations if the changes are ok with you. (I implemented `FromRef for Arc<CoreClient>` instead as it's clone isn't as cheap as cloning an `Arc`.)

**Pull request merged**